src/blockdata/transaction.rs

```diff
@@ -362,6 +383,42 @@ impl Transaction {
         sha256d::Hash::hash(&raw_vec)
     }
 
+    /// Returns the fee for the transaction by substracting the sum of the output amounts from the
+    /// provided set of transactions `spent_txs` outputs matching the transaction inputs. If some of
+    /// the provided transaction outputs inputs is not matching any of the transaction inputs than
+    /// the `UnmatchedTxForInput` error is returned. If a transaction matching current transaction
+    /// input does not have a corresponding output number, then a `UnmatchedVoutForInput` error is
+    /// returned. If some of the transaction outputs are not covered by any outputs within the
+    /// provided `spent_txs`, then `UnmatchedInput` error is returned. Otherwise, the returned value
+    /// is the amount of the transaction fee in satoshis.
+    pub fn get_fee(&self, spent_txs: Vec<Transaction>) -> Result<u64, Error> {
+        let txid = self.txid();
+        let txins: Result<Vec<u64>, Error> = spent_txs.iter().map(|tx| {
+            match self.input.iter().find(|txin| {
+                txin.previous_output.txid == tx.txid()
```

Comment:
Also, calculating `txid()` is quite slow, and the user might have done it already. In here some `txid`s would be calculated multiple times since this loop is nested, increasing the run time.

Reply:
Another argument to not have this functionality inside rust-bitcoin IMO.